src/flows/repo.rs

```diff
@@ -2,25 +2,28 @@ use crate::filesystem;
 use crate::fzf;
 use crate::git;
 use crate::structures::fzf::{Opts as FzfOpts, SuggestionType};
+use anyhow::Context;
+use anyhow::Error;
 use git2::Repository;
-use std::error::Error;
 use std::fs;
 use std::io::Write;
 use walkdir::WalkDir;
 
-pub fn browse() -> Result<(), Box<dyn Error>> {
-    let repo_path_str = format!("{}/featured", filesystem::tmp_path_str());
+pub fn browse() -> Result<(), Error> {
+    let repo_path_str = format!("{}/featured", filesystem::tmp_path_str()?);
 
-    filesystem::remove_dir(&repo_path_str);
-    filesystem::create_dir(&repo_path_str);
+    // The dir might not exist which would throw an error. But here we don't care about that
```

Comment:
Do you suggest another approach? I agree this is very error-prone

Reply:
Checking whether the dir exists before trying to remove it is a bit more robust, but not worth it imho. The check + remove is not atomic so `remove_dir` would still be able to return the same errors. Just less frequently.